qemu/tests/scsi_mq_check.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import utils_misc
+
+
+def get_mq_from_guest(session, params, cmd, pattern):
+    """
+    Get multi queue number from guest.
+
+    :param session: VM session
+    :param params: Dictionary with the test parameters
+    :param cmd: command to get queue info
+    :param pattern: Get the queue number from queue info
+    """
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        return output
+    if params.get("os_type") == "windows":
+        try:
+            queues = int(re.search(pattern, output, re.I | re.M).group(1))
+        except IndexError:
+            return None
+    if params.get("os_type") == "linux":
+        queues = int(len(re.findall(pattern, output, re.I | re.M)))
+    if queues:
+        return queues
+    else:
+        return None
```

Comment:
`get_mq_from_guest` returns `None`, `int` and `str`.
If it returns `None`, it will fall into the line 60 which will be a test error, but I think your original idea is to get a test fail here.

```python
def get_mq_from_guest(test, session, params, cmd, pattern):
    status, output = session.cmd_status_output(cmd)
    if status != 0:
        test.error("Get multi queue info failed, check the detail result:\n%s" % result_guest)
    if params.get("os_type") == "windows":
        try:
            queues = int(re.search(pattern, output, re.I | re.M).group(1))
        except IndexError:
            queues = 0
    if params.get("os_type") == "linux":
        queues = int(len(re.findall(pattern, output, re.I | re.M)))
    return queues
```
so from line 60 to 69 will be:
```python
        if result_guest == int(num_queues):
            error_context.context("The num_queues from guest is match with expected",
                                  logging.info)
        else:
            test.fail("The num_queues from guest is not match with expected.\n"
                      "num_queues from guest is %s, expected is %s"
                      % (result_guest, num_queues))

```

Reply:
Hi hachen, thanks for you comment, if the get_mq_from_guest returns None or str, the line 60 will interrupt the progress, report test.error. 
if not isinstance(result_guest, int):
            test.error("Get multi queue info failed, check the detail result:\n%s"
                       % result_guest)

if the get_mq_from_guest return int, the line 63 will be executed" if result_guest == int(num_queues):"